### PR TITLE
Feature store data handling

### PR DESCRIPTION
**Pull request by Lavhe**
I have full functionality for communicating with all components (Account and Chat)

Soo users can simply call 

```
registerUser(user)
loginUser(user) 
getCurrentUser()
getChats(senderID)
getMsgs(senderID,receiverID)
addMsg(text,senderID,receiverID)
```

all changes are handled in a single store

.... View **src/components/home.js** to see the basic usage

### CODE REVIEW

File: src/components/headerNav.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import Avatar from 'material-ui/Avatar';
+import RaisedButton from 'material-ui/RaisedButton';
+import '../css/index.css'
+
+class HeaderNav extends Component{
+  state = {
+      props:'',
+      logged: false,
+      user:{
+        img:"https://images.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.idyllwildarts.org%2Fwp-content%2Fuploads%2F2016%2F09%2Fblank-profile-picture.jpg&f=1",
```
Comment (jaderabbit) [inline src/components/headerNav.js]: probably store locally

File: src/components/headerNav.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                  <MenuItem primaryText="Log out" onClick={()=>{ this.setState({ logged:false })}}  />
+                </IconMenu>
+                <IconButton>
+                  <Avatar src="https://files.slack.com/files-pri/T02LJS8M9-F8PK0UGEB/icon.png"/>
+                </IconButton>
+              </div>
+              :
+              <div>
+                <Route render={({ history}) => (
+                  <FlatButton {...this.props} onClick={()=>{
```
Comment (jaderabbit) [inline src/components/headerNav.js]: what does this do? `{...this.props}`

File: src/components/headerNav.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                      this.setState({ logged:true })
+                      alert("TODO : Send the user to the login page")
```
Comment (WesleyBuck) [inline src/components/headerNav.js]: Remove alert from pr. just leave TODO Comment

File: src/css/index.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+.rectangle-3 {
+  height: 103px;
+  width: 100%;
+  background-color: #01B9BD;
+}
+
+.login {
+  height: 25px;	width: 146px;	color: #FFFFFF;	font-family: "Avenir Next";	font-size: 18px;	font-weight: 500;	line-height: 25px;	text-align: center;
```
Comment (jaderabbit) [inline src/css/index.css]: fam, no! new lines

File: src/reducers/Account.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      userName: 'Joseph',
+      userSurname: 'Sirwali',
+      userEmail: 'mulavhe@gmail.com',
+      userImg: 'https://images.duckduckgo.com/iu/?u=https%3A%2F%2Fpbs.twimg.com%2Fprofile_images%2F1396485177%2Ftrevor_cartoon_profile.jpg&f=1'
+    }
+  ]
+}
+
+export default(state = initialState, action) => {
+  switch (action.type) {
```
Comment (jaderabbit) [inline src/reducers/Account.js]: :tada: Redux!

File: src/css/index.css [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+.oval {
+  height: 269px;	width: 269px;	background-color: #FFFFFF;	box-shadow: 0 2px 12px 3px rgba(0,0,0,0.14);
+}
+
+.rectangle {
+  height: 45px;	width: 45px;	border: 3px solid #004853;
+}
+
+.addie-hogan-addiehog {
```
Comment (WesleyBuck) [inline src/css/index.css]: ??

File: src/css/index.css [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  border-radius:50%;
+}
+
+.text-center{
+  margin:auto;
+  text-align: center;
+}
+
+.oval {
+  height: 269px;	width: 269px;	background-color: #FFFFFF;	box-shadow: 0 2px 12px 3px rgba(0,0,0,0.14);
```
Comment (WesleyBuck) [inline src/css/index.css]: ???

File: src/components/headerNav.js (hunk 1 above)
Comment (Lavhe) [reply src/components/headerNav.js]: Will read from the store as soon as this branch  is merged with develop

File: src/components/headerNav.js (hunk 2 above)
Comment (Lavhe) [reply src/components/headerNav.js]: It adds all the props to the FlatButton (via it's constructor) soo all its properties can easily merge with the root properties

File: src/css/index.css (hunk 4 above)
Comment (Lavhe) [reply src/css/index.css]: Solved

File: src/reducers/Account.js (hunk 5 above)
Comment (Lavhe) [reply src/reducers/Account.js]: YEY!!!!

File: src/components/headerNav.js (hunk 3 above)
Comment (Lavhe) [reply src/components/headerNav.js]: Oh sorry ,,,, solved

File: src/css/index.css (hunk 7 above)
Comment (Lavhe) [reply src/css/index.css]: Solved

File: src/css/index.css (hunk 6 above)
Comment (Lavhe) [reply src/css/index.css]: Solved

**Pull request merged**